### PR TITLE
feat(plotly): add reactive line selection support

### PR DESCRIPTION
**Pull request by axsseldz**
## 📝 Summary

This PR makes pure Plotly line charts (`scatter/scattergl` with `mode="lines"`) fully selection-reactive in marimo:
- Click now returns selected point data and indices.
- Box selection filters points by both `x` and `y` bounds.
- Lasso selection returns only points inside the polygon.

## 🔍 Description of Changes

- Updated the Plotly frontend plugin to properly handle line-trace click selection:
  - Accept click events for pure line traces (not only heatmaps).
  - Extract indices robustly from `pointIndex`, `pointNumber`, or `pointNumbers`.
  - Recover missing `x/y` values from trace data (`data`/`fullData`) when Plotly emits index-only click payloads.
  - Clear stale selection state (`range`, `lasso`, `selections`) on click/deselect transitions.

- Improved modebar behavior for pure line charts:
  - Ensures box/lasso selection tools are available and usable for line-only traces.
  - Preserves expected dragmode behavior while allowing selection interactions.

- Extended backend Plotly selection conversion:
  - Added scatter/line extraction for box selections using `range.x + range.y` filtering.
  - Added scatter/line extraction for lasso selections using polygon containment.
  - Ensured selected output includes consistent point payloads and indices.

- Added/updated tests for line selection behavior:
  - Click payload passthrough for pure lines.
  - XY box filtering for `scatter` and `scattergl` lines.
  - Edge case: segment crossing a box without vertex inclusion.
  - Lasso selection extraction for pure line traces.
  - Numpy/fallback parity for selection extraction paths.

## 📋 Checklist

- [x] I have read the [contributor guidelines](https://github.com/marimo-team/marimo/blob/main/CONTRIBUTING.md).
- [ ] For large changes, or changes that affect the public API: this change was discussed or approved through an issue, on [Discord](https://marimo.io/discord?ref=pr), or the community [discussions](https://github.com/marimo-team/marimo/discussions) (Please provide a link if applicable).
- [x] Tests have been added for the changes made.
- [x] Documentation has been updated where applicable, including docstrings for API changes.
- [x] Pull request title is a good summary of the changes - it will be used in the [release notes](https://github.com/marimo-team/marimo/releases).

CC: @mscolnick @nojaf

### CODE REVIEW

**Comment by axsseldz** [comment]
@mscolnick just updated this pr with some changes, should be fine right now 👍

File: frontend/src/plugins/impl/plotly/PlotlyPlugin.tsx [hunk 1]
```diff
@@ -91,6 +95,199 @@ const SUNBURST_DATA_KEYS: (keyof Plotly.SunburstPlotDatum)[] = [
...
+
+export function shouldHandleClickSelection(
+  points: readonly Plotly.PlotDatum[],
+): boolean {
+  return points.some(
+    (point) => getTraceSource(point).type === "heatmap" || isLinePoint(point),
+  );
+}
+
+export function extractIndices(points: readonly Plotly.PlotDatum[]): number[] {
```
Comment (mscolnick) [inline frontend/src/plugins/impl/plotly/PlotlyPlugin.tsx]: would it be possible to move many of these new functions/utils to another file and add some tests for it?

Comment (axsseldz) [reply frontend/src/plugins/impl/plotly/PlotlyPlugin.tsx]: @mscolnick yes definitely!, i just moved the new plotly line selection helpers out of PlotlyPlugin.tsx into a plotlySelection.ts module and added frontend tests for them in plotlySelection.test.ts. I also verified the existing backend Plotly selection tests still pass

File: marimo/_plugins/ui/_impl/plotly.py [hunk 2]
```diff
@@ -986,7 +1001,10 @@ def _extract_scatter_points_fallback(
                         cast(Any, y_min_p) <= y_val <= cast(Any, y_max_p)
                     )
                 else:
-                    y_position = y_category_positions.get(y_val)
+                    try:
```
Comment (nojaf) [inline marimo/_plugins/ui/_impl/plotly.py:1037]: This seems a bit sus. Maybe add a code comment on when this can go wrong, useful for future readers.
